src/mode.js

```diff
@@ -43,13 +43,14 @@ let fullVersion = '';
 
 /**
  * Provides info about the current app.
+ * @param {!Window=} win
  * @return {!ModeDef}
  */
-export function getMode() {
+export function getMode(win) {
   if (mode) {
```

Comment:
It's fine since the service worker will get it's own global scope. Are you thinking we should skip the cache if there's an explicit window?